pkg/converter/convert_test.go

```diff
@@ -213,30 +213,49 @@ func TestConvert(t *testing.T) {
 			},
 		},
 		{
-			name:         "exec format kubeconfig azurecli",
-			execArgItems: []string{getTokenCommand, argEnvironment, envName, argServerID, serverID, argClientID, clientID, argTenantID, tenantID, argLoginMethod, "azurecli"},
+			name:         "exec format kubeconfig, convert from azurecli to azurecli",
+			execArgItems: []string{getTokenCommand, argEnvironment, envName, argServerID, serverID, argClientID, clientID, argTenantID, tenantID, argLoginMethod, token.AzureCLILogin},
 			expectedArgs: []string{getTokenCommand, argServerID, serverID, argLoginMethod, token.AzureCLILogin},
 			overrideFlags: map[string]string{
 				flagLoginMethod: token.AzureCLILogin,
 			},
 			command: execName,
 		},
 		{
-			name:          "exec format kubeconfig azurecli with args as overrides",
+			name:         "exec format kubeconfig, convert from azurecli to azurecli, with envName as overrides",
+			execArgItems: []string{getTokenCommand, argServerID, serverID, argClientID, clientID, argTenantID, tenantID, argLoginMethod, token.AzureCLILogin},
+			expectedArgs: []string{getTokenCommand, argServerID, serverID, argLoginMethod, token.AzureCLILogin},
+			overrideFlags: map[string]string{
+				flagLoginMethod: token.AzureCLILogin,
+				flagEnvironment: envName,
+			},
+			command: execName,
+		},
+		{
+			name:          "exec format kubeconfig, convert from azurecli to azurecli, with args as overrides",
 			execArgItems:  []string{getTokenCommand},
 			expectedArgs:  []string{getTokenCommand, argServerID, serverID, argLoginMethod, token.AzureCLILogin},
 			overrideFlags: map[string]string{flagLoginMethod: token.AzureCLILogin, flagServerID: serverID, flagClientID: clientID, flagTenantID: tenantID, flagEnvironment: envName},
 			command:       execName,
 		},
 		{
-			name:         "exec format kubeconfig DeviceCodeLogin",
-			execArgItems: []string{getTokenCommand, argEnvironment, envName, argServerID, serverID, argClientID, clientID, argTenantID, tenantID, argLoginMethod, "azurecli"},
+			name:         "exec format kubeconfig, convert from azurecli to DeviceCodeLogin",
```

Comment:
since the primary conversion is from devicelogin, please add the test matrix covering devicelogin -> the rest of supported login methods